Source/AliveLib/Scrab.cpp

```diff
@@ -627,8 +628,483 @@ __int16 Scrab::AI_Patrol_0_4AA630()
 
 __int16 Scrab::AI_ChasingEnemy_1_4A6470()
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    __int16 result;
+
+    Scrab *pScrabToFight = Scrab::FindScrabToFight_4A4E20();
+    if (pScrabToFight)
+    {
+        SetBrain(&Scrab::AI_Fighting_2_4A5840);
+        field_124_fight_target_obj_id = pScrabToFight->field_8_object_id;
+        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+        result = 0;
+    }
+    else
+    {
+        BaseGameObject *pObj = sObjectIds_5C1B70.Find_449CF0(field_120_obj_id);
+        BaseAliveGameObject *v6 = (BaseAliveGameObject *)pObj;

```

Comment:
get rid of v6 and use `auto pObj = static_cast<BaseAliveGameObject*>(sObjectIds_5C1B70.Find_449CF0(field_120_obj_id));`